x-pack/plugins/apm/public/components/shared/FilterBar/DatePicker.tsx

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import datemath from '@elastic/datemath';
+import {
+  CommonProps,
+  EuiSuperDatePicker,
+  EuiSuperDatePickerProps
+} from '@elastic/eui';
+import { memoize } from 'lodash';
+import React from 'react';
+import { connect } from 'react-redux';
+import { RouteComponentProps, withRouter } from 'react-router-dom';
+import { updateTimePicker } from 'x-pack/plugins/apm/public/store/urlParams';
+import { fromQuery, toQuery } from '../Links/url_helpers';
+
+interface Props extends RouteComponentProps {
+  dispatchUpdateTimePicker: typeof updateTimePicker;
+}
+
+interface DatePickerParams {
+  rangeFrom: string;
+  rangeTo: string;
+  refreshPaused: boolean;
+  refreshInterval: number;
+}
+
+// TODO this can be removed when EUI PR lands: https://github.com/elastic/eui/pull/1574
+declare module '@elastic/eui' {
+  interface OnTimeChangeProps {
+    start: string;
+    end: string;
+    isInvalid: boolean;
+    isQuickSelection: boolean;
+  }
+
+  interface OnRefreshChangeProps {
+    isPaused: boolean;
+    refreshInterval: number;
+  }
+
+  export type EuiSuperDatePickerProps = CommonProps & {
+    start?: string;
+    end?: string;
+    isPaused?: boolean;
+    refreshInterval?: number;
+    onTimeChange: (props: OnTimeChangeProps) => void;
+    onRefreshChange?: (props: OnRefreshChangeProps) => void;
+    showUpdateButton?: boolean;
+  };
+  export const EuiSuperDatePicker: React.SFC<EuiSuperDatePickerProps>;
+}
+
+const APM_DEFAULT_TIME_OPTIONS: DatePickerParams = {
+  rangeFrom: 'now-24h',
+  rangeTo: 'now',
+  refreshPaused: true,
+  refreshInterval: 0
+};
+
+class DatePickerComponent extends React.Component<Props> {
+  public refreshTimeoutId = 0;
+
+  public getParamsFromSearch = memoize((search: string) => {
+    const query = toQuery(search);
+    const converted = {
+      ...query,
+      refreshPaused:
+        query.refreshPaused === 'true' || query.refreshPaused === true,
```

Comment:
Value will always be a string, right?
```js
refreshPaused: query.refreshPaused === 'true'
```

This is one of the reasons I do all conversion in urlParams eg:
https://github.com/jasonrhodes/kibana/blob/e6f30167d67f030c180b098d5f3a4d20fa15622b/x-pack/plugins/apm/public/store/urlParams.ts#L61

The rest of the application only uses `urlParams` (and not the raw url values) and thus can be sure they are the correct type.

Reply:
Yeah, I mean I'm not totally against using urlParams for all of this. I think it may cause a little confusion around start/end vs to/from, but if we think that's the simplest way to handle all these values I can probably get behind it. What do you think?